src/ui/graphicsview.cpp

```diff
@@ -16,15 +17,68 @@ void GraphicsView::mouseReleaseEvent(QMouseEvent *event) {
     QGraphicsView::mouseReleaseEvent(event);
 }
 
-void GraphicsView::drawForeground(QPainter *painter, const QRectF&) {
-    for (auto item : this->overlay.getItems()) {
-        item->render(painter);
-    }
-}
-
 void GraphicsView::moveEvent(QMoveEvent *event) {
     QGraphicsView::moveEvent(event);
     QLabel *label_MapRulerStatus = findChild<QLabel *>("label_MapRulerStatus", Qt::FindDirectChildrenOnly);
     if (label_MapRulerStatus && label_MapRulerStatus->isVisible())
         label_MapRulerStatus->move(mapToGlobal(QPoint(6, 6)));
 }
+
+void MapView::drawForeground(QPainter *painter, const QRectF&) {
+    foreach (Overlay * overlay, this->overlayMap)
```

Comment:
Is this guaranteed to iterate over the `QMap`'s keys in ascending order?  I'm having trouble finding documentation for that.

Reply:
`QMap`s are sorted by key
https://doc.qt.io/qt-5/qmap.html#details